### PR TITLE
DAOS-8741-test2: rebuild/container_create_race test failed due to pool create -1011 timeout and -1012” pool-busy on the race condition

### PR DESCRIPTION
**Pull request by dinghwah**
Description: Update test for rebuild container race condition.
            Test steps:
            (1)Start IOR before rebuild.
            (2)Starting rebuild by killing rank.
            (3)Wait for rebuild to start.
            (4)Race condition, create containers, data write/read during
               rebuild.
            (5)Wait for rebuild to finish.
            (6)Check for pool and rebuild info after rebuild.
modified:   container_create_race.py
modified:   container_create_race.yaml
Test-tag: pr rebuild_cont_create
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### CODE REVIEW

**Comment by brianjmurrell** [comment]
@dinghwah Have you merged this PR per this e-mail sent out earlier this week requested/required?

```
From:		"Murrell, Brian" <brian.murrell@intel.com>
Reply-To:	"Murrell, Brian" <brian.murrell@intel.com>
To:		daos-devel@eclists.intel.com <daos-devel@eclists.intel.com>
Subject:	[daos-devel] ACTION REQUIRED: Please rebase master and release/2.0 ASAP
Date:		Sat, 06 Nov 2021 19:39:04 +0000 (2021-11-06 03:39:04 PM)

Patches have landed to both master and release/2.0 that will be
required in your PRs for them to pass after a pipeline-lib PR lands
early on Monday.

Please rebase/merge any PRs before pushing any more commits to GitHub,
to avoid your PR from failing in test.

Cheers,
b.
```

File: src/tests/ftest/rebuild/container_create_race.py [hunk 1]
```diff
@@ -1,284 +1,166 @@
...
         Returns:
             bool: was the opening and closing of the container successful
 
         """
         status = True
+        container = self.container[index]
         self.log.info(
-            "%s: Verifying the container %s created during rebuild",
-            loop_id, message)
+            "..Vrifying the container %s created during rebuild", container)
```
Comment (shimizukko) [inline src/tests/ftest/rebuild/container_create_race.py:66]:
```suggestion
            "..Verifying the container %s created during rebuild", container)
```
typo

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/rebuild/container_create_race.py (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/rebuild/container_create_race.py]: Thanks for the review.
Please check the update.

**Review by shimizukko — APPROVED** [review]

**Comment by dinghwah** [comment]
Rebuild/container_create_race test is stable and passed in this PR.
The failure on pr-head (pool/create_capacity failed during teardown) is irrelated to this PR.

File: src/tests/ftest/rebuild/container_create_race.py [hunk 2]
```diff
@@ -1,284 +1,166 @@
...
-
     Test Class Description:
         These rebuild tests verify the ability to create additional containers
         while rebuild is ongoing.
-
     :avocado: recursive
     """
 
-    def add_containers_during_rebuild(self, loop_id, qty, pool1, pool2):
+    def add_containers_during_rebuild(self, qty=10, index=-1):
```
Comment (rpadma2) [inline src/tests/ftest/rebuild/container_create_race.py:21]: Any reason you used index 1? We could have used default index=0 for iterating the container array... Is it not?

File: src/tests/ftest/rebuild/container_create_race.py [hunk 3]
```diff
@@ -1,284 +1,166 @@
...
-            loop_id (str): loop identification string
             index (int): index of the daos container object to open/close
-            message (str): additional text describing the container
-
         Returns:
             bool: was the opening and closing of the container successful
 
         """
         status = True
+        container = self.container[index]
```
Comment (rpadma2) [inline src/tests/ftest/rebuild/container_create_race.py:64]: Same as above.
Any reason you used index 1? We could have used default index=0 for iterating the container array... Is it not?

File: src/tests/ftest/rebuild/container_create_race.py [hunk 4]
```diff
@@ -1,284 +1,166 @@
...
+        self.log.info("..(6)Check for pool and rebuild info after rebuild")
+        info_checks["pi_ndisabled"] += targets
+        rebuild_checks["rs_obj_nr"] = ">0"
+        rebuild_checks["rs_rec_nr"] = ">0"
+        self.assertTrue(
+            self.pool.check_pool_info(**info_checks),
+            "Invalid pool information detected before rebuild")
+        self.assertTrue(
+            self.pool.check_rebuild_status(**rebuild_checks),
+            "Invalid pool rebuild info detected after rebuild")
```
Comment (rpadma2) [inline src/tests/ftest/rebuild/container_create_race.py:166]: How sure we are the data migrated by rebuild didn't lose data? Maybe not critical... But there is a daos container (data integrity) check which can be run to make sure rebuild did the job correctly.

File: src/tests/ftest/rebuild/container_create_race.py (hunk 2 above)
Comment (dinghwah) [reply src/tests/ftest/rebuild/container_create_race.py:21]: Thanks for the review and comment.
"index -1" was chosen for the latest container created by default.

File: src/tests/ftest/rebuild/container_create_race.py (hunk 3 above)
Comment (dinghwah) [reply src/tests/ftest/rebuild/container_create_race.py:64]: same as above

File: src/tests/ftest/rebuild/container_create_race.py (hunk 4 above)
Comment (dinghwah) [reply src/tests/ftest/rebuild/container_create_race.py:166]: This testcase is to check the container create during restart, same as the other rebuild testcases, the pool and rebuild info counters are verified before and after the rebuild, data integrity is check with ior and io write and read.

Comment (rpadma2) [reply src/tests/ftest/rebuild/container_create_race.py:166]: I understand. How do you know rebuild performed it job correctly? Doing a data integrity check with make sure the data is not corrupted after rebuild.

**Review by rpadma2 — COMMENTED** [review]
LGTM.

I would recommend to have a separate ticket to enhance the test case to validate the data integrity of the objects after rebuild is complete.

**Review by rpadma2 — APPROVED** [review]

**Pull request merged**